plugins/UsersManager/API.php

```diff
@@ -1317,19 +1315,14 @@ public function createAppSpecificTokenAuth($userLogin, $md5Password, $descriptio
             }
         }
         
-        if (empty($user) || !$this->password->verify($md5Password, $user['password'])) {
+        if (empty($user) || !$this->passwordVerifier->isPasswordCorrect($userLogin, $passwordConfirmation)) {
             /**
              * @ignore
              * @internal
              */
             Piwik::postEvent('Login.authenticate.failed', array($userLogin));
```

Comment:
btw @diosmosis the login failed event should be only triggered when `empty($user)` since `isPasswordCorrect` will already  trigger `Piwik::postEvent('Login.recordFailedLoginAttempt');` which was the goal of triggering that event. Maybe could even simply remove the `empty($user)` then we no longer need to post that event at all? Assuming the authenticate method should fail anyway when no user is found.

Reply:
So `Login.recordFailedLoginAttempt` and `Login.authenticate.failed` are the same?

Reply:
![image](https://user-images.githubusercontent.com/273120/81884465-1d066400-95ec-11ea-9c92-e03525887cc7.png)
pretty much yes. Would have actually been better and more clear initially to use `Login.recordFailedLoginAttempt` there.